src/main/java/org/apache/commons/crypto/cipher/CipherFactory.java

```diff
@@ -47,6 +50,9 @@
    */
   public static Cipher getInstance(CipherTransformation transformation,
       Properties props) throws GeneralSecurityException {
+    if(props == null){
```

Comment:
I would suggest to put the check of props == null into Utils.getCipherClasses method. 
This can centralize the point of handling null props and avoid the need of creating a new Properties object.